### PR TITLE
feat: Add gha for merging prod into main

### PR DESCRIPTION
**Pull request by alldoami**
## Summary

Step in release action to create a PR after release steps are done. I caught up with Masoud and he said this is the only manual step that is still happening in the release process.

Here is some fun documentation on the `GITHUB_TOKEN` secret that I learned: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

Here is documentation about the create pr gha: https://github.com/marketplace/actions/create-pull-request

Added the ability to autoapprove and automerge the PR. Docs: https://github.com/marketplace/actions/enable-pull-request-automerge.

Did thorough testing here: https://github.com/chanzuckerberg/ghas-internal-feature-testing/pull/7. Thanks to @hspitzley-czi for helping me out with this!

**Feedback I'd like:**
Is this the right place to put this? Should this step go at the very bottom of this file after job 3? I put it at the end of Job 1 because if all of those steps succeed, then this should be run, correct?

## Checklist

- [ ] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [ ] Test Story in Storybook
- [ ] Tests written
- [ ] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by tihuan — COMMENTED** [review]
Hurray! Thanks so much for closing the manual loop in the deploy process for us, @alldoami 🤩 👏 

Really appreciate the context shared in the PR description too. Super helpful and lots of TILs 💡 

Will wait for @masoudmanson to LGTM, in case I missed anything lol

Thanks again!!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -90,34 +90,53 @@ jobs:
...
+      # Step 14: Generate token for autoapproving PR
+      - name: Generate token
+        id: generate_token
+        uses: chanzuckerberg/github-app-token@v1.1.4
+        with:
+          app_id: ${{ secrets.CZI_GITHUB_HELPER_APP_ID }}
+          private_key: ${{ secrets.CZI_GITHUB_HELPER_PK }}
 
-      # Step 14: Autoapprove PR
+      # Step 15: Autoapprove PR
```
Comment (tihuan) [inline .github/workflows/release.yml:135]: Brilliant 🔥 !!! Thanks so much for auto merging the PR 🚀 💡 This is SO exciting!!

**Review by tihuan — COMMENTED** [review]
Woahhh can't wait to get this merged!! Thanks so much again for making this possible, @alldoami 🤩 🙏 You BEYOND amazing!!

**Review by tihuan — APPROVED** [review]
LGTM!! 🚀 Thanks so much again!!

File: .github/workflows/release.yml (hunk 1 above)
Comment (masoudmanson) [reply .github/workflows/release.yml:135]: Incredible work, @alldoami! Thank you for your dedication and effort in putting this together. it's truly remarkable! 😍🥳🤩

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -89,6 +89,55 @@ jobs:
...
+      # Step 14: Generate token for autoapproving PR
+      - name: Generate token
+        id: generate_token
+        uses: chanzuckerberg/github-app-token@v1.1.4
+        with:
+          app_id: ${{ secrets.CZI_GITHUB_HELPER_APP_ID }}
+          private_key: ${{ secrets.CZI_GITHUB_HELPER_PK }}
+
+      # Step 15: Autoapprove PR
+      - name: Autoapprove PR
```
Comment (masoudmanson) [inline .github/workflows/release.yml:136]: Oh 😯
I had no idea we could auto-approve the PRs ⭐️
This is fantastic! Thanks for sharing 🥳

**Review by masoudmanson — APPROVED** [review]
Thank you, @alldoami, for your contribution. This means a lot to us! 😍 I'm going to go ahead and merge this PR and start using the new workflows. 🥳

**Pull request merged**